### PR TITLE
fixed header parsing in language server

### LINKED ISSUE #34667: Incorrect parsing of language server request headers
**Godot version:**
3.2.beta

**OS/device including version:**
archlinux x86_64

**Issue description:**
Starting a message to the language server websocket with "Content-Length:" makes the server ignore the message.  This header is required by the the current lsp specification(3.14) and so it seems very odd to ignore it.  I believe this may be a shortcut taken to make it easier for the only current client implementation (the one for vscode) to communicate, but because it does not comply to the specification it is very confusing for a developer who is attempting to implement another client (I am currently trying to write one in emacs for lsp-mode, which sends headers the correct way).

**Steps to reproduce:**
Send a message to the language server (using a tool like websocat) starting with a "Content-Length: <size>" header.  There will be no output.

### PR DESCRIPTION
**Pull request by ofrank123**
Remove headers in language server requests to serve as a fix for the issue mentioned in issue #34667 

*Bugsquad edit:* Fixes #34667.

### CODE REVIEW

File: modules/gdscript/language_server/gdscript_language_protocol.cpp [hunk 1]
```diff
@@ -43,6 +43,8 @@ void GDScriptLanguageProtocol::on_data_received(int p_id) {
 	if (OK == peer->get_packet_buffer(data)) {
 		String message;
 		message.parse_utf8((const char *)data.read().ptr(), data.size());
+		std::size_t json_start = message.find("{");
```
Comment (akien-mga) [inline modules/gdscript/language_server/gdscript_language_protocol.cpp]: The build fails https://ci.appveyor.com/project/akien-mga/godot/builds/29805069

The type is also wrong, `String::find` returns an `int`, not a `size_t`.

Comment (Geequlim) [reply modules/gdscript/language_server/gdscript_language_protocol.cpp]: We'd better follow the LSP specification.

> The header part consists of header fields. Each header field is comprised of a name and a value, separated by ‘: ‘ (a colon and a space). Each header field is terminated by ‘\r\n’. Considering the last header field and the overall header itself are each terminated with ‘\r\n’, and that at least one header is mandatory, this means that two ‘\r\n’ sequences always immediately precede the content part of a message.

You can get the content body start position by `message.find("\r\n\r\n")`

**Comment by Geequlim** [comment]
You'd better test does it break the VSCode and atom plugin.

**Comment by ofrank123** [comment]
I've tested on VSCodium (the OSS version), and it seems to work fine.  I agree that as it stands the LSP spec is not faithfully implemented, but doing so would break the current VSCode plugin, which I think more people probably use than my hacky emacs set up.  There's a few things I plan on trying to implement other than just this header parsing (first priority is the missing support for textDocument/didSave).

**Comment by akien-mga** [comment]
> I agree that as it stands the LSP spec is not faithfully implemented, but doing so would break the current VSCode plugin, which I think more people probably use than my hacky emacs set up.

Well if that's the case, I think the VSCode plugin should be fixed. @Geequlim is its main dev so they can do it.
We haven't had any stable release yet with LSP support, so we can still change the server and clients to better fit the specification.

**Comment by Geequlim** [comment]
> I agree that as it stands the LSP spec is not faithfully implemented, but doing so would break the current VSCode plugin, which I think more people probably use than my hacky emacs set up.

If you find something else is not following the LSP standard let me know.

**Comment by akien-mga** [comment]
> You can get the content body start position by `message.find("\r\n\r\n")`

Ping.

**Comment by ofrank123** [comment]
If I make that change should I change it over to raw TCP instead of websockets?  Otherwise there's no reason to have a Content-Length header.  TCP is a much more common way of doing language servers anyways, so I think it would benefit people trying to build clients for GDScript.  This would require changes to the vscode's languge server client, but I don't think it would be that difficult.

**Comment by NathanLovato** [comment]
Other language servers seem to communicate over TCP. At least, Emacs's LSP protocol support expects communication over TCP and does not support websocket at the moment. Maybe some other code editors might be in the same situation and would expect language server clients to use TCP for communication?

**Comment by akien-mga** [comment]
Closing as it's not being updated as requested to follow the specification.

**Pull request closed without merging**